cpp/src/community/leiden_impl.cuh

```diff
@@ -604,26 +604,21 @@ void relabel_cluster_ids(raft::handle_t const& handle,
                          size_t num_nodes)
 {
   vertex_t local_cluster_id_first{0};
+
+  // Get unique cluster id and shuffle
+  remove_duplicates<vertex_t, multi_gpu>(handle, unique_cluster_ids);
+
   if constexpr (multi_gpu) {
-    auto unique_cluster_range_lasts = cugraph::partition_manager::compute_partition_range_lasts(
-      handle, static_cast<vertex_t>(unique_cluster_ids.size()));
-
-    auto& comm                 = handle.get_comms();
-    auto const comm_size       = comm.get_size();
-    auto const comm_rank       = comm.get_rank();
-    auto& major_comm           = handle.get_subcomm(cugraph::partition_manager::major_comm_name());
-    auto const major_comm_size = major_comm.get_size();
-    auto const major_comm_rank = major_comm.get_rank();
-    auto& minor_comm           = handle.get_subcomm(cugraph::partition_manager::minor_comm_name());
-    auto const minor_comm_size = minor_comm.get_size();
-    auto const minor_comm_rank = minor_comm.get_rank();
-
-    auto vertex_partition_id =
-      partition_manager::compute_vertex_partition_id_from_graph_subcomm_ranks(
-        major_comm_size, minor_comm_size, major_comm_rank, minor_comm_rank);
-
-    local_cluster_id_first =
-      vertex_partition_id == 0 ? vertex_t{0} : unique_cluster_range_lasts[vertex_partition_id - 1];
+    auto cluster_ids_size_per_rank = cugraph::host_scalar_allgather(
+      handle.get_comms(), unique_cluster_ids.size(), handle.get_stream());
+
+    std::vector<vertex_t> cluster_ids_starts(cluster_ids_size_per_rank.size());
+    std::exclusive_scan(cluster_ids_size_per_rank.begin(),
+                        cluster_ids_size_per_rank.end(),
+                        cluster_ids_starts.begin(),
+                        size_t{0});
+    auto& comm             = handle.get_comms();
```

Comment:
Looks good, functionally.

If you're going to pull out `&comm` from the handle, I'd do that at line 612 (right as you enter the if block and use `comm` also in the call to `host_scalar_allgather`.  Otherwise I would just use `handle.get_comms().get_rank()` as the index in the next line.

But that's purely cosmetic.